core/plugins/org.polarsys.capella.core.data.gen.edit.decorators/src/org/polarsys/capella/core/data/interaction/provider/SequenceMessageItemProviderDecorator.java

```diff
@@ -29,8 +33,37 @@ public SequenceMessageItemProviderDecorator(AdapterFactory adapterFactory) {
 		super(adapterFactory);
 	}
 
-  @Override
-  public String getText(Object object) {
-    return super.getText(object);
-  }
+    @Override
+    public String getText(Object object) {
+        return super.getText(object);
+    }
+
+    @Override
+    public Object getImage(Object object) {
+        MessageKind kind = ((SequenceMessage) object).getKind();
+        
+        final String imagePath; //$NON-NLS-1$
+        switch (kind) {
+        case REPLY:
+            imagePath = "full/obj16/SequenceMessageReply"; //$NON-NLS-1$
+            break;
+        case CREATE:
+            imagePath = "full/obj16/SequenceMessageCreate"; //$NON-NLS-1$
+            break;
+        case DELETE:
+            imagePath = "full/obj16/SequenceMessageDelete"; //$NON-NLS-1$
+            break;
+        case TIMER:
+            if (((SequenceMessage) object).getReceivingEnd().getEvent() instanceof ArmTimerEvent) {

```

Comment:
can you do some null checks on getReceivingEnd as it occured many times that messages was badly formed.

Reply:
Done